test/cmdlineTests/mcopy_bytes_array_returned_from_function/input.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.0.0;
+
+contract C {
+    function foo() external pure returns (bytes memory)
+    {
+        bytes memory ret = "aaaaa";
+        return ret;
```

Comment:
I don't think this bit is even necessary to trigger the `copyToMemoryFunction()` code path. It's just the fact that the function is external and returns `bytes memory`, which means there's ABI encoding needed. The body can be left empty to make the output slightly shorter.

Reply:
By the way, I see two code paths that use `copyToMemoryFunction()` and here's a snippet that would trigger the other one (decoding a byte array stored in memory):

```solidity
contract C {
    function foo() external pure {
        abi.decode("abcd", (bytes));
    }
}
```

Reply:
I've added this test case in a [separate commit](https://github.com/ethereum/solidity/pull/14834/commits/e931a8e7882643f5f765461b6477ed50be677803) so you can take a look. There's no `mcopy` in there (or rather much of anything) since the whole thing likely gets optimized away as unused (not the `mcopy` op by itself, rather the `memory_to_memory_copy_....` function), which is why I pushed it without the `--optimize` flag. I could capture the return value from the `abi.decode(...)` call and return it, but that would essentially just be a duplicate of the other test, which would make it redundant.